packages/rum-recorder/src/domain/rrweb/record.spec.ts

```diff
@@ -52,7 +54,7 @@ describe('record', () => {
     expect(records.filter((record) => record.type === RecordType.FullSnapshot).length).toEqual(1)
   })
 
-  it('is safe to checkout during async callbacks', (done) => {
+  it('is safe to checkout during async callbacks (old mutation observer)', (done) => {
```

Comment:
does checkout mean snapshot?
there is a isCheckout parameter for takeFullSnapshot but is it used?

Reply:
Mmh the 'checkout' concept have been moved outside of our RRWeb codebase. It was a way to take full snapshots at regular intervals or when a given number of records have been emitted.

In the [upstream RRWeb test](https://github.com/rrweb-io/rrweb/blob/master/test/record.test.ts#L168) they specify a "checkout" every 2 records, so it occured exactly after `span.innerText = 'test'`. Since we removed this concept from the RRWeb codebase, I used a `takeFullSnapshot` call to emulate the behavior.

I think the test is here because the RRWeb author feared that mutation occured while taking a full snapshot (hence the "freeze" concept removed in https://github.com/DataDog/browser-sdk/pull/802).

Because we know that taking a full snapshot is synchronous, "async callbacks" will never be a problem. Also, we'll have a much better code coverage of mutations, so we may not need this test anymore. I'd be in favor of removing it. WDYT?

The `isCheckout` parameter is not used indeed, and can be removed.

Reply:
🔥 🔥 🔥

Reply:
`isCheckout` argument removed in https://github.com/DataDog/browser-sdk/pull/820